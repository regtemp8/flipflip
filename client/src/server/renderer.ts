import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
