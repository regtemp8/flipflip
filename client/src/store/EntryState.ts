export interface Identifiable {
  id: number
}

export interface EntryState<T extends Identifiable> {
  name: string
  nextID: number
  entries: Record<number, T>
}

export interface EntryUpdate<T> {
  id: number
  value: T
}

export function isEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
) {
  return state.entries[id] != null
}

export function setNextID<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (state.nextID <= entry.id) {
    state.nextID = entry.id + 1
  }
}

export function getEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
): T {
  return state.entries[id]
}

export function setEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  state.entries[entry.id] = entry
  setNextID(state, entry)
}

export function createEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (!isEntry(state, entry.id)) {
    state.entries[entry.id] = entry
    setNextID(state, entry)
  } else {
    throw new Error(
      `${state.name}: Create failed. Entry with ID ${entry.id} already exists.`
    )
  }
}

export function createEntries<T extends Identifiable>(
  state: EntryState<T>,
  entries: T[]
) {
  for (const entry of entries) {
    if (isEntry(state, entry.id)) {
      throw new Error(
        `${state.name}: Create failed. Entry with ID ${entry.id} already exists.`
      )
    }
  }

  entries.forEach((s) => (state.entries[s.id] = s))
  const maxID = Math.max(...entries.map((s) => s.id))
  setNextID(state, { id: maxID })
}

export function updateEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (isEntry(state, entry.id)) {
    state.entries[entry.id] = entry
  } else {
    throw new Error(
      `${state.name}: Update failed. Entry with ID ${entry.id} doesn't exist.`
    )
  }
}

export function deleteEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
) {
  if (isEntry(state, id)) {
    delete state.entries[id]
  } else {
    throw new Error(
      `${state.name}: Delete failed. Entry with ID ${id} doesn't exist.`
    )
  }
}
