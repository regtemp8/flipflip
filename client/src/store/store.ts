import { configureStore } from '@reduxjs/toolkit'
import appSlice from './app/slice'
import audioSlice, { initialAudioState } from './audio/slice'
import constantsSlice from './constants/slice'
import playersSlice, { initialPlayersState } from './player/slice'
import captionScriptSlice, {
  initialCaptionScriptState
} from './captionScript/slice'
import clipSlice, { initialClipState } from './clip/slice'
import librarySourceSlice, {
  initialLibrarySourceState
} from './librarySource/slice'
import overlaySlice, { initialOverlayState } from './overlay/slice'
import playlistSlice, { initialPlaylistState } from './playlist/slice'
import sceneSlice, { initialSceneState } from './scene/slice'
import sceneGridSlice, { initialSceneGridState } from './sceneGrid/slice'
import sceneGroupSlice, { initialSceneGroupState } from './sceneGroup/slice'
import tagSlice, { initialTagState } from './tag/slice'
import videoClipperSlice, {
  initialVideoClipperState
} from './videoClipper/slice'
import scenePickerSlice, { initialScenePickerState } from './scenePicker/slice'
import captionScriptorSlice, {
  initialCaptionScriptorState
} from './captionScriptor/slice'
import sceneDetailSlice, { initialSceneDetailState } from './sceneDetail/slice'
import audioLibrarySlice, {
  initialAudioLibraryState
} from './audioLibrary/slice'
import { initialApp } from './app/data/App'
import { initialSystemConstants } from './constants/SystemConstants'
import sourceScraperSlice, {
  initialSourceScraperState
} from './sourceScraper/slice'

const store = configureStore({
  reducer: {
    app: appSlice,
    audio: audioSlice,
    captionScript: captionScriptSlice,
    clip: clipSlice,
    constants: constantsSlice,
    librarySource: librarySourceSlice,
    overlay: overlaySlice,
    playlist: playlistSlice,
    scene: sceneSlice,
    sceneGrid: sceneGridSlice,
    sceneGroup: sceneGroupSlice,
    tag: tagSlice,
    // components
    players: playersSlice,
    videoClipper: videoClipperSlice,
    scenePicker: scenePickerSlice,
    captionScriptor: captionScriptorSlice,
    sceneDetail: sceneDetailSlice,
    audioLibrary: audioLibrarySlice,
    sourceScraper: sourceScraperSlice
  }
})

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch