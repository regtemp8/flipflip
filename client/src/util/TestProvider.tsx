import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
