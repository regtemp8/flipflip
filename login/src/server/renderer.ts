import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
