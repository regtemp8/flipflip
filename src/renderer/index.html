<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
