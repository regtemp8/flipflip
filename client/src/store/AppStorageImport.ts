import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { EntryState } from "./EntryState";
import type Tag from "./tag/Tag";
import type SceneGroup from "./sceneGroup/SceneGroup";
import type SceneGrid from "./sceneGrid/SceneGrid";
import type Scene from "./scene/Scene";
import type Playlist from "./playlist/Playlist";
import type Overlay from "./overlay/Overlay";
import type LibrarySource from "./librarySource/LibrarySource";
import type Clip from "./clip/Clip";
import type CaptionScript from "./captionScript/CaptionScript";
import type Audio from "./audio/Audio";
import type App from "./app/data/App";

export interface AppStorageImport {
  app: App;
  audio: EntryState<Audio>;
  captionScript: EntryState<CaptionScript>;
  clip: EntryState<Clip>;
  librarySource: EntryState<LibrarySource>;
  overlay: EntryState<Overlay>;
  playlist: EntryState<Playlist>;
  scene: EntryState<Scene>;
  sceneGrid: EntryState<SceneGrid>;
  sceneGroup: EntryState<SceneGroup>;
  tag: EntryState<Tag>;
}

export const initialAppStorageImport: AppStorageImport = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
}