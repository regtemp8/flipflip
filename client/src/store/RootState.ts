import { RootState } from "./store";
import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialSystemConstants } from "./constants/SystemConstants";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { initialPlayersState } from "./player/slice";
import { initialVideoClipperState } from "./videoClipper/slice";
import { initialScenePickerState } from "./scenePicker/slice";
import { initialCaptionScriptorState } from "./captionScriptor/slice";
import { initialSceneDetailState } from "./sceneDetail/slice";
import { initialAudioLibraryState } from "./audioLibrary/slice";
import { initialSourceScraperState } from "./sourceScraper/slice";

export const initialRootState: RootState = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  constants: initialSystemConstants,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
  players: initialPlayersState,
  videoClipper: initialVideoClipperState,
  scenePicker: initialScenePickerState,
  captionScriptor: initialCaptionScriptorState,
  sceneDetail: initialSceneDetailState,
  audioLibrary: initialAudioLibraryState,
  sourceScraper: initialSourceScraperState
}