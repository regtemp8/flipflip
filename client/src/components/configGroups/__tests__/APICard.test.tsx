import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import APICard from "../APICard";
import TestProvider from "../../../util/TestProvider";
import store from "../../../store/store";

jest.mock("../../library/SourceIcon", () => "SourceIcon");

describe("APICard", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider store={store}>
        <APICard />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
