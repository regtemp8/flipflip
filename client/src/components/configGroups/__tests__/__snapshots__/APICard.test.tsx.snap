// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APICard should match snapshot 1`] = `
Array [
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter APICard-title-7 css-1uwgr7b-MuiTypography-root"
  >
    API Sign In
  </p>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-919fip-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        aria-label="Unauthorized: Click to Authorize Tumblr"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default APICard-fab-2 APICard-noAuth-4 css-mm42gz-MuiButtonBase-root-MuiFab-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <SourceIcon
          className="APICard-icon-5"
          type="st.tumblr"
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        aria-label="Unauthorized: Click to Authorize Instragram"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default APICard-fab-2 APICard-noAuth-4 css-mm42gz-MuiButtonBase-root-MuiFab-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <SourceIcon
          className="APICard-icon-5"
          type="st.instagram"
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        aria-label="Unauthorized: Click to Configure Hydrus"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default APICard-fab-2 APICard-noAuth-4 css-mm42gz-MuiButtonBase-root-MuiFab-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <SourceIcon
          className="APICard-icon-5"
          type="st.hydrus"
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        aria-label="Unauthorized: Click to Configure Piwigo"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default APICard-fab-2 APICard-noAuth-4 css-mm42gz-MuiButtonBase-root-MuiFab-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <SourceIcon
          className="APICard-icon-5"
          type="st.piwigo"
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 APICard-center-9 css-1idn90j-MuiGrid-root"
    >
      <label
        className="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            aria-disabled={true}
            className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-disabled Mui-disabled PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-disabled Mui-disabled css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              disabled={true}
              onChange={[Function]}
              required={false}
              type="checkbox"
            />
            <span
              className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
          </span>
          <span
            className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-ahj2mt-MuiTypography-root"
        >
          Silence Tumblr Throttle Alert
        </span>
      </label>
    </div>
  </div>,
]
`;
